Tests/MongoKittenTests/CRUDTests.swift

```diff
@@ -1,404 +1,168 @@
+        concern.acknowledgement = .majority
+    }
+    
+    override func tearDownWithError() throws {
+        try mongo.drop().wait()
+    }
+    
+    func printDummyAccounts () throws {
+        let schema = mongo[DummyAccount.collectionName]
+        try schema.find().forEach{ dummy in
+            print("\(dummy["_id"]!): Name - \(dummy["name"]!) | password - \(dummy["password"]!) | age - \(dummy["age"]!)")
+        }.wait()
+    }
+    
+    func executeCommand<E, R>(command: E, replyType: R.Type) -> EventLoopFuture<R> {
+        <#function body#>
+    }
+    
+    func testCreateDummyAccount () throws {
+        let schema = mongo[DummyAccount.collectionName]
+        let startingCount = try schema.count().wait()
+        
+        let dummyAccount = DummyAccount(name: "Dum", password: "my", age: 69)
+        let _ = try schema.insertEncoded(dummyAccount).wait()
+        XCTAssertEqual(try schema.count().wait(), startingCount+1)
+    }
+    
+    func testBulkCreateDummyAccounts () throws {
+        let dummyAccounts = [
+            DummyAccount(name: "Test", password: "ing", age: 77),
+            DummyAccount(name: "To", password: "see", age: 8),
+            DummyAccount(name: "If", password: "bulk", age: 10),
+            DummyAccount(name: "Inserts", password: "will", age: 15),
+            DummyAccount(name: "Work", password: "as", age: 19),
+            DummyAccount(name: "I", password: "expect", age: 30),
+            DummyAccount(name: "Them", password: "to", age: 82),
+        ]
+        
+        let documents = try dummyAccounts.map { dummyAccount in
+            try BSONEncoder().encode(dummyAccount)
+        }
+        
+        var command = InsertCommand(documents: documents, inCollection: DummyAccount.collectionName)
+        command.writeConcern = concern
+        let connection = try mongo.pool.next(for: .init(writable: false)).wait()
+        let insertReply = try connection.executeCodable(command, namespace: mongo.commandNamespace, in: nil, sessionId: mongo.sessionId).decodeReply(InsertReply.self).wait()
+        
+        XCTAssertEqual(insertReply.ok, 1)
+        XCTAssertEqual(insertReply.insertCount, dummyAccounts.count)
+    }
+    
+    func readDummyAccount (name : String) throws -> DummyAccount? {
+        let schema = mongo[DummyAccount.collectionName]
+        return try schema.findOne("name" == name, as: DummyAccount.self).wait()
+    }
+    
+    func testReadDummyAccounts () throws {
+        try testBulkCreateDummyAccounts()
+        if let dummy = try readDummyAccount(name: "Them") {
+            XCTAssertEqual(dummy.password, "to")
+            XCTAssertEqual(dummy.age, 82 )
+        } else {
+            XCTFail("Retrieved a nil value")
+        }
+    }
+    
+    func testBulkReadDummyAccounts () throws {
+        try testBulkCreateDummyAccounts()
+        let schema = mongo[DummyAccount.collectionName]
+        var dummyCounter = 0
+        
+        try schema.find().forEach{ dummy in
```

Comment:
Use `.allResults()` here. Within `forEach` the value cannot be `nil`, so the assertion is unnecessary.

`let results = schema.find().allResults()`

Then you can compare `results.count` rather than `dummyCounter`